packages/neon-core/src/sc/manifest/ContractMethodDescriptor.ts

```diff
@@ -0,0 +1,41 @@
+import { ContractParamType } from "../ContractParam";
+import {
+  ContractEventDescriptor,
+  ContractEventDescriptorLike
+} from "./ContractEventDescriptor";
+import { ContractParameterDefinition } from "./ContractParameterDefinition";
+
+export interface ContractMethodDescriptorLike
+  extends ContractEventDescriptorLike {
+  returnType: ContractParamType;
+}
+
+export class ContractMethodDescriptor extends ContractEventDescriptor {
```

Comment:
generally I would advise against this inheritance if we are just sharing 2 fields. Is there a need to cast the method descriptor to a event descriptor?

Reply:
I just copied this from neo core code.
As far as I can see, they are just sharing 2 fields.

Reply:
Let's split it and not use inheritance then.

Reply:
OK